src/cascadia/TerminalSettingsModel/FolderEntry.cpp

```diff
@@ -43,6 +49,72 @@ winrt::com_ptr<NewTabMenuEntry> FolderEntry::FromJson(const Json::Value& json)
     JsonUtils::GetValueForKey(json, NameKey, entry->_Name);
     JsonUtils::GetValueForKey(json, IconKey, entry->_Icon);
     JsonUtils::GetValueForKey(json, EntriesKey, entry->_Entries);
+    JsonUtils::GetValueForKey(json, InliningKey, entry->_Inlining);
+    JsonUtils::GetValueForKey(json, AllowEmptyKey, entry->_AllowEmpty);
 
     return entry;
 }
+
+// A FolderEntry should only expose the entries to actually render to WinRT,
+// to keep the logic for collapsing/expanding more centralised.

```

Comment:
I'm a little bit hesitant with this method, actually. Right now, we're giving two masters control over when entries are visible: `TerminalPage` makes its own decisions for each type of entry independently from this one, except when it asks a `Folder` and then `Folder` makes decisions and reports back to `Page`. Now, this doesn't hide any of the complexity from `Page`: it still needs to check whether a folder is empty, inlined, or forced to display as empty.

I'm not sure that this is more centralized. :smile:

Reply:
I also don't like this. In a way, it's just view logic. But then it's also partly functional logic. Putting all the filtering in the `TerminalPage` felt like a bad idea, since we also can't access the implementation types there, but putting view logic in the models also felt dirty. Any ideas on how to fix this?